Use vue and api to handle workflow imports

guerler opened the pull request:
Currently the import workflow form is linked to a legacy controller endpoint and uses a backbone view. This PR adds the form as vue component linked to the api. It also properly registers the 'myexperiment' configuration parameter. This avoids requesting it through an additional ajax request. Additionally errors are displayed instantly without reloading the grid, this allows users to fix them without having to re-navigate to the import form. This PR also fixes importing workflows from myexperiment which currently does not work.

guerler committed:
Add routes for workflow import view, apply consistent routes

guerler committed:
Add vue component for workflow import form

guerler committed:
Remove legacy view

guerler committed:
Add import workflow from file option to api endpoint

guerler committed:
Remove unused controller endpoint, and disabled legacy option

guerler committed:
Remove unnecessary check

guerler committed:
Restore broken installed_repository_file option

guerler committed:
Remove unnecessary spacing and unused variable

nsoranzo commented on lib/galaxy/webapps/galaxy/api/workflows.py:
```diff
@@ -266,26 +267,44 @@ def create(self, trans, payload, **kwd):
+            installed_repository_file = payload.get('installed_repository_file', '')
+            workflow_file = open(installed_repository_file, 'rb')
+            workflow_data = workflow_file.read()
+            workflow_file.close()
```
Can you use the `with` statement instead of explicitly closing the file, please?

nsoranzo commented on lib/galaxy/webapps/galaxy/api/workflows.py:
```diff
@@ -266,26 +267,44 @@ def create(self, trans, payload, **kwd):
+                else:
+                    raise exceptions.MessageException("You attempted to upload an empty file.")
+            else:
+                raise exceptions.MessageException("Please provide a url or file.")
```
s/url/URL/

guerler replied:
Thanks (:

guerler committed:
Fix library usage, and parameter naming

Fix file reading caller

Avoid using bare except

Fix file parameter name

Add missing library

guerler committed:
Add additional checks for installed repository handler, open files using with-block

guerler committed:
Use archive_source and not url

guerler committed:
Fix error message

guerler committed:
Show warning if import was successful but has errors

guerler committed:
Fix spacing in import block

guerler committed:
Attempt to fix selenium tests

Fix lint

guerler committed:
Check tool version when importing subworkflows from dictionary

guerler committed:
Always match exact tool version

guerler committed:
Allow upgrade when workflow is loaded into editor

guerler committed:
Allow tool version upgrade for the run form

guerler committed:
Remove redundant default for tool version flag

guerler committed:
Return proper error message if module cannot be produced

guerler committed:
Add additional check for section handler

guerler committed:
Apply consistent group value handling

guerler committed:
Adjust test case

guerler committed:
Merge branch 'exact_version_handling' into workflow_import

guerler committed:
Fix failing tests

guerler committed:
Restore error handling variation

guerler committed:
Remove api flag check from workflow manager

guerler committed:
Add source to workflow name

guerler committed:
Fix import from myexperiment service

guerler committed:
Fix default parameters for 3rd party workflow import controller

guerler committed:
Remove unused variable

guerler committed:
Remove unused source identifier

jmchilton commented:
Merged it locally against dev to test the form with a branch including BS4, it looks good still. Thanks @guerler.
